CMakeLists.txt

```diff
@@ -55,11 +54,18 @@ option(DLAF_BUILD_DOC "Build documentation" OFF)
 option(DLAF_WITH_PRECOMPILED_HEADERS "Use precompiled headers." OFF)
 option(DLAF_WITH_SCALAPACK "Build ScaLAPACK-like C API (requires ScaLAPACK)" OFF)
 
-if(DLAF_WITH_MKL AND DLAF_WITH_MKL_LEGACY)
-  message(SEND_ERROR "DLAF_WITH_MKL and DLAF_WITH_MKL_LEGACY are mutually exclusive.")
-endif()
+try_compile(
+  DLAF_WITH_MKL_TRY_COMPILE "${PROJECT_SOURCE_DIR}/cmake/tests"
+  "${PROJECT_SOURCE_DIR}/cmake/tests/mkl_set_num_threads.cpp"
+  CMAKE_FLAGS "-DINCLUDE_DIRECTORIES=${LAPACK_INCLUDE_DIR}"
+  LINK_LIBRARIES ${LAPACK_LIBRARY}
+)
+# TODO: Maybe don't need to print this? Though maybe it's nice to have...
```

Comment:
To do.

Reply:
Something like:
`Intel MKL detection: [Not ]Found`?

Reply:
Works for me. `message(STATUS)` is probably sufficient? Or do you think it's useful to make the not found case `message(WARNING)`? I'd worry it's too noisy if one isn't actually interested in using MKL.

Reply:
I agree with both of you. A **STATUS** message like **"Intel MKL: (NOT) Found"** saying if MKL has been found or not.

Reply:
I think a warning would break CP2K CI.

Reply:
Changed it to a status message.